File: Problem3.rb
require 'prime'
big_num, dest = 600851475143, 0
roof = Math.sqrt(big_num).floor
roof.downto(0) do |i|
  if big_num % i == 0 && Prime.prime?(i)
    dest = i
    break
  end
  break if dest != 0  
end
puts dest

File: Problem3.js
function isPrime(n){
  if( n%1 || n<2 ) return false;
  if (n%2==0) return (n==2);
  if (n%3==0) return (n==3);
  var roof = Math.sqrt(n);
  for (i=5; i<= roof; i+=6){
    if(n%i==0) return false;
    if(n%(i+2)==0) return false;
  }
  return true;
}

function largestPrimeFactor(num){
  var roof = Math.floor(Math.sqrt(num))
  for (i=roof; i > 0; i--){
    if (num%i == 0 && isPrime(num)){
      return i;
    }
    console.log(i);
  }
}

console.log(largestPrimeFactor(600851475143))
